tests/test_workspaces.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use itertools::Itertools;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+/// Test adding a second workspace
+#[test]
+fn test_workspaces_add_second_workspace() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "--git", "main"]);
+    let main_path = test_env.env_root().join("main");
+    let secondary_path = test_env.env_root().join("secondary");
+
+    std::fs::write(main_path.join("file"), "contents").unwrap();
+    test_env.jj_cmd_success(&main_path, &["close", "-m", "initial"]);
+
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @"default: 988d8c1dca7e 
+");
```

Comment:
Nit: I know very little about rust's strings/raw strings. Can we get this "); off the left edge somehow, by either replacing the expected value's newline with a \n, or using r### strings? Or would that break the insta review feature?

Reply:
`insta` can deal with extra indentation, but I've found that when I change the indentation level later, `rustfmt` doesn't adjust the raw string literals to match (because that would be a behavioral change), so then it looks funny later anyways. At this point, I just ignore the formatting and let `insta` manage it.

Reply:
Yes, unless `rustfmt` can be configured to change to indent it for us, I don't think it's worth going through them and manually changing after running `cargo insta test --review`.